package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,266 @@
+
+from .base import AnalysisBase
+
+logger = logging.getLogger("MDAnalysis.analysis.diffusionmap")
+
+
+class DiffusionMap(AnalysisBase):
+    """Non-linear dimension reduction method
+
+    Dimension reduction with diffusion map of the structures in the universe.
+
+    Attributes
+    ----------
+    atoms : AtomGroup
+        Selected atoms in trajectory subject to dimension reduction
+    diffusion_matrix : array
+        Array of all possible ij metric distances between frames in trajectory
+    eigenvalues: array
+        Eigenvalues of the diffusion map
+    eigenvectors: array
+        Eigenvectors of the diffusion map
+        The second and higher eigenvectors ev[i+1,:] represent the i-th slowest
+        collective coordinates.
+
+    """
+
+    def __init__(self, u,  select='all', epsilon='average', k=10, weights=None,
+                 metric=None, start=None, stop=None, step=None):
+        """
+        Parameters
+        -------------
+        u : trajectory `~MDAnalysis.core.AtomGroup.Universe`
+            The MD Trajectory for dimension reduction, remember that computational
+            cost scales at O(N^3) where N is the number of frames.
+            Cost can be reduced by increasing step interval or specifying a
+            start and stop
+        select: str, optional
+            1. any valid selection string for
+            :meth:`~MDAnalysis.core.AtomGroup.AtomGroup.select_atoms`
+            This selection of atoms is used to calculate the RMSD between
+            different frames. Water should be excluded.
+        epsilon : float, optional
+            Specifies the epsilon used for the diffusion map. More information
+            in [1] and [2]. With 'average' the average of the RMSD to the k-th
+            greatest value is used. (#TODO, check if actually agrees with theory)
+        k : int, optional
+            specifies the k for the epsilon choice if average epsilon is
+            used.
+        weights: list, optional
+            The list has to have the same length as the trajectory.
+            With 'None' the weight of each frame of the trajectory will be the
+            same.
+        metric : function, optional
+            Maps two numpy arrays to a float, is positive definite and symmetric,
+            Default: ``None`` sets metric to qcp.CalcRMSDRotationalMatrix
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+        """
+        self._u = u
+        self.select = select
+        self.atoms = u.select_atoms(select)
+        self._natoms = self.atoms.n_atoms
+
+        traj = u.trajectory
+        self._epsilon = epsilon
+        if metric is not None:
+            self._metric = metric
+        else:
+            self._metric = qcp.CalcRMSDRotationalMatrix
```

Comment:
please use the wrapped RMSD functions and provide some way to set options for the metric functions. 
The qcp function expects the inputs to be preprocessed (that's why you need casting in `_single_frame`). Rather use the `rms.rmsd` function and provide a way to specify arguments for the metric function.